common/src/main/java/gov/cms/ab2d/common/service/CoverageServiceImpl.java

```diff
@@ -84,57 +103,162 @@ private CoverageSearchEvent getLastEvent(CoveragePeriod period) {
     }
 
     @Override
-    public CoverageSearchEvent insertCoverage(int searchId, long searchEventId, Collection<String> beneficiaryIds) {
+    public CoverageSearchEvent insertCoverage(int periodId, long searchEventId, List<String> beneficiaryIds) {
 
-        CoveragePeriod period = findCoveragePeriod(searchId);
+        // Make sure that coverage period and searchEvent actually exist in the database before inserting
+        findCoveragePeriod(periodId);
         CoverageSearchEvent searchEvent = findCoverageSearchEvent(searchEventId);
 
-        Collection<Coverage> coverages = beneficiaryIds.stream()
-                .map(bene -> buildCoverage(period, searchEvent, bene)).collect(toList());
+        int written = 0;
+        while (written < beneficiaryIds.size()) {
```

Comment:
This would be better as a for loop, all the information on one line ... including the increment later in the loop.